File: Capitulo4/secao_4_2/Python/sockets/TCP/TCPServer.py
import socket
import threading

class Connection(threading.Thread):
    def __init__(self, client_socket, client_address):
        threading.Thread.__init__(self)
        self.client_socket = client_socket
        self.client_address = client_address
        self.start()

    def run(self):
        try:
            data = self.client_socket.recv(1024).decode('utf-8')
            print(f"Received from {self.client_address}: {data}")
            self.client_socket.sendall(data.upper().encode('utf-8'))
        except Exception as e:
            print(f"Error with client {self.client_address}: {e}")
        finally:
            self.client_socket.close()

def main():
    host = '0.0.0.0'  # Escuta em todas as interfaces
    port = 7896

    try:
        print("Servidor iniciado")
        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as server_socket:
            server_socket.bind((host, port))
            server_socket.listen()

            while True:
                client_socket, client_address = server_socket.accept()
                print(f"Conexão estabelecida com {client_address}")
                Connection(client_socket, client_address)

    except Exception as e:
        print("Listen socket error:", e)

if __name__ == "__main__":
    main()

File: Capitulo4/secao_4_2/Java/sockets/TCP/TCPServer.java
package secao_4_2.sockets.TCP;

import java.net.*;
import java.io.*;

public class TCPServer {
	public static void main(String args[]) {
		try {
			System.out.println("Servidor iniciado");
			int serverPort = 7896; // the server port
			ServerSocket listenSocket = new ServerSocket(serverPort);
			while (true) {
				Socket clientSocket = listenSocket.accept();
				System.out.println(clientSocket.getInetAddress());
				System.out.println("conexão estabelecida");
				Connection c = new Connection(clientSocket);
			}
		} catch (IOException e) {
			System.out.println("Listen socket:" + e.getMessage());
		}
	}
}

class Connection extends Thread {
	DataInputStream in;
	DataOutputStream out;
	Socket clientSocket;

	public Connection(Socket aClientSocket) {
		try {
			clientSocket = aClientSocket;
			in = new DataInputStream(clientSocket.getInputStream());
			out = new DataOutputStream(clientSocket.getOutputStream());
			this.start();
		} catch (IOException e) {
			System.out.println("Connection:" + e.getMessage());
		}
	}

	public void run() {
		try { // an echo server

			String data = in.readUTF(); // read a line of data from the stream
			System.out.println(data);
			out.writeUTF(data.toUpperCase());
		} catch (EOFException e) {
			System.out.println("EOF:" + e.getMessage());
		} catch (IOException e) {
			System.out.println("readline:" + e.getMessage());
		} finally {
			try {
				clientSocket.close();
			} catch (IOException e) {
				/* close failed */}
		}

	}
}